samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -365,7 +365,7 @@ def _add_account_id_to_global(self) -> None:
                 if self._global_parameter_overrides is None:
                     self._global_parameter_overrides = {}
                 self._global_parameter_overrides["AWS::AccountId"] = account_id
-        except (NoCredentialsError, TokenRetrievalError, ClientError):
+        except (Exception):
             LOG.warning("No current session found, using default AWS::AccountId")
```

Comment:
Not blocking anything, but I wonder if its worth logging what exception we voided as a debug log. Since this method is a "best effort" attempt at getting an account ID, its not critical, but could help with visibility into what happened.